### PR TITLE
Add support for NIOTransportServices

### LINKED ISSUE #29: Network.framework support
Hi, how could I use this websocket on iOS client app without entire Vapor stack? I'm looking for alternative light weight SocketIO implementation

### PR DESCRIPTION
**Pull request by PopFlamingo**
This PR intends to fix #29 by using the NIOTransportServices library

### CODE REVIEW

File: Sources/WebSocketKit/WebSocketClient.swift [hunk 1]
```diff
@@ -44,7 +45,11 @@ public final class WebSocketClient {
         case .shared(let group):
             self.group = group
         case .createNew:
+            #if canImport(Network)
+            self.group = NIOTSEventLoopGroup(loopCount: 1, defaultQoS: .default)
+            #else
             self.group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+            #endif
```
Comment (gwynne) [inline Sources/WebSocketKit/WebSocketClient.swift:52]: I would leave this part out - there are legitimate reasons to want to default to a plain MTELG even when Network.framework is available (most often on macOS). The `.shared` case provides ample ability to specify an appropriate ELG if desired. (Worth noting, I suggest it as much for consistency with other packages that do the same thing as for any other reason.)

**Review by gwynne — CHANGES_REQUESTED** [review]
One note, other than that, looks great!

**Comment by gwynne** [comment]
N.B.: Marked `semver-minor`, as the added dependency on NIOTransportServices is noteworthy in semver terms.

File: Sources/WebSocketKit/WebSocketClient.swift (hunk 1 above)
Comment (PopFlamingo) [reply Sources/WebSocketKit/WebSocketClient.swift:52]: @gwynne I see, thanks! Will remove it in next commit 👍

File: Package.swift [hunk 2]
```diff
@@ -4,14 +4,15 @@ import PackageDescription
 let package = Package(
     name: "websocket-kit",
     platforms: [
-       .macOS(.v10_15)
+        .macOS(.v10_15), .watchOS(.v6)
```
Comment (gwynne) [inline Package.swift:7]:
```suggestion
        .macOS(.v10_15),
        .iOS(.v13),
        .watchOS(.v6),
        .tvOS(.v13),
```

**Review by gwynne — CHANGES_REQUESTED** [review]

**Comment by 0xTim** [comment]
@PopFlamingo if you can fix the merge conflicts I'll get this merged

**Review by gwynne — DISMISSED** [review]

**Comment by PopFlamingo** [comment]
@0xTim @gwynne If you are fine with it I will also drop the watchOS support for now, in my tests it wasn't possible to make it work on Apple Watch because of OS restrictions. IIRC Apple documentation states that some types of apps such as music streaming apps have lower level networking access but I couldn't test it.

**Comment by 0xTim** [comment]
@PopFlamingo fine by me - we can always add it in if someone really wants to make it work

**Review by 0xTim — APPROVED** [review]

**Pull request merged**